superset-frontend/src/dashboard/components/RefreshIntervalModal.tsx

```diff
@@ -91,6 +99,51 @@ class RefreshIntervalModal extends React.PureComponent<
     this.setState({
       refreshFrequency: value || refreshIntervalOptions[0][0],
     });
+
+    if (value === -1) {
+      this.setState({
+        custom_block: true,
+      });
+    } else {
+      this.setState({
+        custom_block: false,
+      });
+    }
+  }
+
+  onSaveValue(value: number) {
+    this.props.onChange(value, this.props.editMode);
+    this.modalRef?.current?.close();
+    this.props.addSuccessToast(t('Refresh interval saved'));
+  }
+
+  createIntervalOptions(refreshIntervalOptions: [number, string][]) {
+    const refresh_options = [];
+    if (refreshIntervalOptions.length === 0) {
+      refresh_options.push({ value: -1, label: 'Custom interval' });
+      return refresh_options;
+    }
+    refresh_options.push({
+      value: refreshIntervalOptions[0][0],
+      label: t(refreshIntervalOptions[0][1]),
+    });
+    refresh_options.push({ value: -1, label: 'Custom interval' });
+    for (let i = 1; i < refreshIntervalOptions.length; i += 1)
+      refresh_options.push({
+        value: refreshIntervalOptions[i][0],
+        label: t(refreshIntervalOptions[i][1]),
+      });
+    return refresh_options;
+  }
```

Comment:
I think you need to review these logics here. Are you trying to put the custom interval at a specific position of the array?

Reply:
Yes i kept custom options in the array options that we are showing in dropdown.

Reply:
Ok then please consider using `map` rather than the for loop to make the code more readable and concise. Thank you